apps/explorer/src/types/assets.ts

```diff
@@ -16,39 +16,63 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { TokenInfo } from '@alephium/token-list'
-import { FungibleTokenMetaData, NFTMetaData } from '@alephium/web3'
+import { NFTTokenUriMetaData } from '@alephium/shared'
+import { FungibleTokenMetadata, NFTMetadata as NFTMetadataBase } from '@alephium/web3/dist/src/api/api-explorer'
 
 import client from '@/api/client'
 
 export type AssetType = Awaited<ReturnType<typeof client.node.guessStdTokenType>>
 
 export type AssetBase = { id: string; type: AssetType }
 
-export type FungibleTokenMetadata = Omit<FungibleTokenMetaData, 'totalSupply'> & { id: string; verified: boolean }
+export type FungibleTokenMetadataBase = Omit<FungibleTokenMetadata, 'totalSupply' | 'decimals'> & {
+  verified: boolean
+  decimals: number
+  logoURI?: string
+}
+
+export type VerifiedFungibleTokenMetadata = TokenListTokenInfo & {
+  type: 'fungible'
+  verified: true
+}
+
+export type UnverifiedFungibleTokenMetadata = FungibleTokenMetadataBase & { type: 'fungible'; verified: false }
 
-export type VerifiedFungibleTokenMetadata = TokenInfo & { type: 'fungible'; verified: true }
+export type NFTMetadata = NFTMetadataBase & { id: string; type: 'non-fungible'; file?: NFTFile; verified: boolean }
```

Comment:
The `NFTMetadataBase` (basically, the `NFTMetadata` from the explorer API) already has the `id: string` prop: https://github.com/alephium/alephium-web3/blob/master/packages/web3/src/api/api-explorer.ts#L213